app/lib/homework/homework_dialog/homework_dialog.dart

```diff
@@ -337,7 +337,8 @@ class HwDialogErrorStrings {
 }
 
 class _SaveButton extends StatelessWidget {
-  const _SaveButton({this.editMode = false});
+  bool _homeworkIsSaving = false;
+  _SaveButton({this.editMode = false});
 
```

Comment:
Since it's a stateless class you can't (or you shouldn't) use a mutable variable. This causes a warning to be shown, which is why the analyzer CI step is failing. I'll make a recommendation on how to fix this in another comment.

Reply:
Fixed